ci/check-protobuf-stability.sh

```diff
@@ -35,42 +51,76 @@ function check_non_lf_protos() {
-  # from previous release branches (say, against both `1.17.1` and `1.16.2`
-  # created after the `release/1.17.x` branch).
-  # Instead, we approximate by checking only against the most recent stable tag
-  # reachable from the current branch, under the assumption that if a lesser
-  # release branch contains a protobuf change, then it will also be present in
-  # higher ones either through a shared commit or a back-port from `main`.
+}
+
+case "${1:---stable}" in
+-h | --help)
+  cat <<USAGE
+Usage: ./check-protobuf-stability.sh [options]
+
+Options:
+  -h, --help: shows this help
+  --stable:     check against the latest stable tag (default)
+  --target:     check against the tip of the target branch
+USAGE
+  exit
+  ;;
+--stable)
+  # Check against the most recent stable tag.
   #
-  # Finally, this check does not need to run on release branch commits because
+  # This check does not need to run on release branch commits because
   # they are built sequentially, so no conflicts are possible and the per-PR
   # check is enough.
+  readonly RELEASE_BRANCH_REGEX="^release/.*"
   GIT_TAG_SCOPE=""
+  # For PRs targeting release branches, we should really check against
+  # all the most recent stable tags reachable from either the current branch or
+  # from previous release branches (say, against both 1.17.1 and 1.16.2
+  # created after the release/1.17.x branch).
+  #
+  # Check against the most recent stable (non-snapshot) tag reachable from the current branch,
+  # under the assumption that if a previous release branch contains a protobuf change,
+  # then it will also be present in former ones (previous/former with regards to versioning).
   if [[ "${TARGET}" =~ ${RELEASE_BRANCH_REGEX} ]]; then
     GIT_TAG_SCOPE="--merged"
   fi
-
   readonly LATEST_STABLE_TAG="$(git tag ${GIT_TAG_SCOPE} | grep -v "snapshot" | sort -V | tail -1)"
-  echo "Checking protobufs against tag '${LATEST_STABLE_TAG}'"
-  for buf_module in "${BUF_MODULES_AGAINST_STABLE[@]}"; do
-    (eval "$(dev-env/bin/dade assist)" ; buf breaking --config "${buf_module}" --against ".git#tag=${LATEST_STABLE_TAG}")
-  done
-
-}
+  # The current stable release is v1.17 and it includes the buf config file with the default name `buf.yml`.
+  # Once we have the v1.18 release we can remove this check for the updated config (KVL-1131)
+  if [[ $LATEST_STABLE_TAG =~ "v1.17."* ]]; then
+    BUF_CONFIG_UPDATED=false
+  else
+    BUF_CONFIG_UPDATED=true
+  fi
+  BUF_GIT_TARGET_TO_CHECK=".git#tag=${LATEST_STABLE_TAG}"
+  ;;
+--target)
+  # Check against the tip of the target branch.
+  #
+  # This check ensures that backwards compatibility is never broken on the target branch,
+  # which is stricter than guaranteeing compatibility between release tags.
+  #
+  # The files are always split for versions > 1.17, and there is no way of opening a PR against a target <= 1.17 which includes this check
+  BUF_CONFIG_UPDATED=true
+  BUF_GIT_TARGET_TO_CHECK=".git#branch=origin/${TARGET}"
```

Comment:
Why is it important to refer to `origin`?

Reply:
On CI we don't have any local branch, so we need to include the remote as well for buf to understand what we mean.